datalad/ui/dialog.py

```diff
@@ -332,29 +332,50 @@ def get_progressbar(self, *args, **kwargs):
                 *args, frontend=self._tqdm_frontend, **kwargs)
 
 
-# poor man thingie for now
-@auto_repr
-class UnderAnnexUI(DialogUI):
+class UnderAnnexMixIn(object):
+    """Mix-in for UnderAnnex UI backends
+
+    In order for the super() calls to work out, this should come first in
+    inheritance order:
+
+    class MyAnnexBackend(UnderAnnexMixIn, WhatEverElse)
+
+    This should lead to super().x() calling WhatEverElse.x()
+    """
+
     def __init__(self, specialremote=None, **kwargs):
         if 'out' not in kwargs:
             # to avoid buffering
             # http://stackoverflow.com/a/181654/1265472
             #kwargs['out'] = os.fdopen(sys.stderr.fileno(), 'w', 0)
             # but wasn't effective! sp kist straogjt for now
             kwargs['out'] = sys.stderr
-        super(UnderAnnexUI, self).__init__(**kwargs)
+        super().__init__(**kwargs)
         self.specialremote = specialremote
 
     def set_specialremote(self, specialremote):
         self.specialremote = specialremote
 
     def get_progressbar(self, *args, **kwargs):
         if self.specialremote:
+            # Report progress via git-annex
+            from .progressbars import AnnexSpecialRemoteProgressBar
             kwargs = kwargs.copy()
-            kwargs['backend'] = 'annex-remote'
             kwargs['remote'] = self.specialremote
-        return super(UnderAnnexUI, self).get_progressbar(
-                *args, **kwargs)
+            return AnnexSpecialRemoteProgressBar(*args, **kwargs)
+        else:
+            return super().get_progressbar(*args, **kwargs)
+
+
+# poor man thingie for now
```

Comment:
in what sense it is `poor`? looks legit to me

Reply:
Not cleaned up, yet. That's your comment on the (moved) implementation of `UnderAnnexUI` ;-)

Reply:
ah, LOL ;) so it is one of the skeletons ! ;)